hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/AbstractCommitWatcher.java

```diff
@@ -62,7 +62,7 @@ abstract class AbstractCommitWatcher<BUFFER> {
 
   private final XceiverClientSpi client;
 
-  private long totalAckDataLength;
+  private volatile long totalAckDataLength;
```

Comment:
consider using AtomicLong instead. It supports addAndGet() which is exactly what we need.